src/lints/trait_associated_type_added.ron

```diff
@@ -0,0 +1,59 @@
+SemverQuery(
+    id: "trait_associated_type_added",
+    human_readable_name: "non-sealed public trait added associated type without default value",
+    description: "A non-sealed trait has gained an associated type without a default value, which breaks downstream implementations of the trait",
+    required_update: Major,
+    lint_level: Deny,
+    reference_link: Some("https://doc.rust-lang.org/reference/items/associated-items.html#associated-types"),
```

Comment:
This is a better reference link, since it explains the "trait item without default" rule directly. The link that defines associated types doesn't mention that, so it's less useful.
```suggestion
    reference_link: Some("https://doc.rust-lang.org/cargo/reference/semver.html#trait-new-item-no-default"),
```